tool/speed.cc

```diff
@@ -24,23 +24,15 @@
 #include <stdlib.h>
 #include <string.h>
 
-#include <openssl/aead.h>
-#include <openssl/aes.h>
-#include <openssl/bn.h>
-#include <openssl/curve25519.h>
-#include <openssl/crypto.h>
-#include <openssl/digest.h>
-#include <openssl/err.h>
-#include <openssl/ec.h>
-#include <openssl/ecdsa.h>
-#include <openssl/ec_key.h>
-#include <openssl/evp.h>
-#include <openssl/hrss.h>
-#include <openssl/mem.h>
-#include <openssl/nid.h>
-#include <openssl/rand.h>
-#include <openssl/rsa.h>
-#include <openssl/trust_token.h>
+#include "internal.h"
+
+#if !defined(OPENSSL_BENCHMARK)
+#include "bssl_bm.h"
+#include <../crypto/ec_extra/internal.h>
```

Comment:
```suggestion
#include "../crypto/ec_extra/internal.h"
```

Reply:
Implementing this change gives us following error message during build:
`FAILED: tool/CMakeFiles/bssl_bm.dir/speed.cc.o 
/usr/bin/c++  -DBORINGSSL_DISPATCH_TEST -DBORINGSSL_HAVE_LIBUNWIND -DBORINGSSL_IMPLEMENTATION -I../third_party/googletest/include -I/home/ubuntu/dev/boringssl/include -I/home/ubuntu/dev/awslc/include/internal/tool -DAWS_LC_STDALIGN_AVAILABLE -DAWS_LC_BUILTIN_SWAP_SUPPORTED -Wvla -Werror -Wformat=2 -Wsign-compare -Wmissing-field-initializers -Wwrite-strings -ggdb -Wall -fvisibility=hidden -fno-common -Wno-free-nonheap-object -Wimplicit-fallthrough -Wmissing-declarations -std=c++11 -fno-exceptions -fno-rtti -Wshadow -fPIE -MD -MT tool/CMakeFiles/bssl_bm.dir/speed.cc.o -MF tool/CMakeFiles/bssl_bm.dir/speed.cc.o.d -o tool/CMakeFiles/bssl_bm.dir/speed.cc.o -c ../tool/speed.cc
In file included from /home/ubuntu/dev/boringssl/include/openssl/stack.h:62:0,
                 from /home/ubuntu/dev/boringssl/include/openssl/ex_data.h:114,
                 from /home/ubuntu/dev/boringssl/include/openssl/ec_key.h:75,
                 from /home/ubuntu/dev/boringssl/include/openssl/ec.h:391,
                 from ../tool/bssl_bm.h:11,
                 from ../tool/speed.cc:30:
../tool/../crypto/ec_extra/../fipsmodule/ec/internal.h:95:23: error: expected string-literal before ‘bn__small_functions_not_usable’
                       bn__small_functions_not_usable)
                       ^
/home/ubuntu/dev/boringssl/include/openssl/type_check.h:69:62: note: in definition of macro ‘OPENSSL_STATIC_ASSERT’
 #define OPENSSL_STATIC_ASSERT(cond, msg) static_assert(cond, msg)
                                                              ^~~
../tool/../crypto/ec_extra/../fipsmodule/ec/internal.h:95:23: error: expected ‘)’ before ‘bn__small_functions_not_usable’
                       bn__small_functions_not_usable)
                       ^
/home/ubuntu/dev/boringssl/include/openssl/type_check.h:69:62: note: in definition of macro ‘OPENSSL_STATIC_ASSERT’
 #define OPENSSL_STATIC_ASSERT(cond, msg) static_assert(cond, msg)
                                                              ^~~
In file included from ../tool/../crypto/ec_extra/internal.h:20:0,
                 from ../tool/speed.cc:31:
../tool/../crypto/ec_extra/../fipsmodule/ec/internal.h:103:1: error: expected ‘;’ before ‘typedef’
 typedef union {
 ^~~~~~~
`

Reply:
Hmm something is up with the static assert macro `OPENSSL_STATIC_ASSERT`.

Reply:
Okay I see what is going on. You are including the definition of `OPENSSL_STATIC_ASSERT` from BoringSSL (through `bssl_bm.h`). But you are also including internal header files from AWS-LC that expects the AWS-LC version of `OPENSSL_STATIC_ASSERT`. The definition of `OPENSSL_STATIC_ASSERT` in AWS-LC is not compatible with the BoringSSL definition, and hence you get the above errors.

One clue is that you see the following line in the error output:
```
../tool/../crypto/ec_extra/../fipsmodule/ec/internal.h:103:1: error: expected ‘;’ before ‘typedef’ typedef union { ^~~~~~~
```
This can't be the AWS-LC version because it doesn't use a union. It uses a struct: https://github.com/awslabs/aws-lc/blob/main/include/openssl/type_check.h#L90.

Where do you expect this the header file to come from?

You will have the same problem with the line below:
```
#include "../crypto/trust_token/internal.h"
```
if it starts defining things that is not compatible between AWS-LC and BoringSSL.

Reply:
So instead of changing `<../crypto/ec_extra/internal.h>` to `"../crypto/ec_extra/internal.h"` wouldn't it make more sense for me to change `"../crypto/trust_token/internal.h"` to `"<../crypto/trust_token/internal.h>`? Or is there some convention against that?